h2o-algos/src/main/java/hex/tree/TreeStats.java

```diff
@@ -38,4 +38,18 @@ public String toString() {
            ", _mean_depth=" + _mean_depth +
            '}';
   }
+  
+  public void mergeWith(TreeStats otherTreeStats) {
+    if (otherTreeStats._min_depth < this._min_depth) this._min_depth = otherTreeStats._min_depth;
+    if (otherTreeStats._max_depth > this._max_depth) this._max_depth = otherTreeStats._max_depth;
+    if (otherTreeStats._min_leaves < this._min_leaves) this._min_leaves = otherTreeStats._min_leaves;
+    if (otherTreeStats._max_leaves > this._max_leaves) this._max_leaves = otherTreeStats._max_leaves;
+    this._byte_size += otherTreeStats._byte_size;
+    this._num_trees += otherTreeStats._num_trees;
+    this._sum_depth += otherTreeStats._sum_depth;
+    this._sum_leaves += otherTreeStats._sum_leaves;
+    this._num_trees += otherTreeStats._num_trees;
+    _mean_depth = ((float) _sum_depth / _num_trees);
```

Comment:
this code is copy-pasted from the `updateBy` method, I would rather put it into a `private void updateMeans()` also a unit test for this method would be nice

Reply:
fac8196